src/geolocation/component.js

```diff
@@ -110,37 +112,55 @@ function Controller(
 
   $element.on('click', this.toggleTracking.bind(this));
 
-  // @ts-ignore: scope ...
-  const map = $scope.getMapFn();
-  if (!(map instanceof olMap)) {
-    throw new Error('Wrong map type');
-  }
-
   /**
    * @type {angular.IScope}
    */
   this.$scope_ = $scope;
 
+  /**
+   * @type {import("ngeo/message/Notification.js").MessageNotification}
+   */
+  this.notification_ = ngeoNotification;
+
+  /**
+   * @type {import("ngeo/map/FeatureOverlayMgr.js").FeatureOverlayMgr}
+   */
+  this.ngeoFeatureOverlayMgr_ = ngeoFeatureOverlayMgr;
+
+  /**
+   * @type {angular.gettext.gettextCatalog}
+   */
+  this.gettextCatalog_ = gettextCatalog;
+
   /**
    * @type {import("ol/Map.js").default}
    */
-  this.map_ = map;
+  this.map;
```

Comment:
Why removing the _?

Reply:
Because with ``bindToController: true``, the ``map``` attribute of the directive defined at https://github.com/camptocamp/ngeo/blob/bf6fdd7fdab4e56f8adcf73099ffc9481fbaceea/src/geolocation/component.js#L77 is automatically set to ``this.map``. Not ``this.map_``.